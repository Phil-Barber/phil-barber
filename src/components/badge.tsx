import React from 'react';
import styled from 'styled-components';

const StyledBadge = styled.button<{ readonly isSelected: boolean }>`
  border: none;
  outline: none;
  cursor: pointer;
  font-size: 12px;
  display: inline-block;
  width: fit-content;
  padding: 0 ${({ theme }) => theme.spacing.xSmall};
  border-radius: ${({ theme }) => theme.spacing.large};
  margin-right: ${({ theme }) => theme.spacing.xxSmall};

  ${({ isSelected, theme }) =>
    isSelected
      ? `
    background-color: ${theme.colors.darkGrey1};
    color: ${theme.colors.lightGrey1};
  `
      : `
    background-color: ${theme.colors.lightGrey1};
    color: ${theme.colors.darkGrey1};
  `}
`;

interface Props {
  children: React.ReactNode;
  className?: string;
  isSelected?: boolean;
  onClick?: () => void;
}

export const Badge: React.FC<Props> = ({
  children,
  className,
  isSelected = false,
  onClick,
}: Props) => (
  <StyledBadge className={className} isSelected={isSelected} onClick={onClick}>
    {children}
  </StyledBadge>
);
