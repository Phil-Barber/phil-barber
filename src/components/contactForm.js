import React from 'react'
import { useFormik } from 'formik'
import * as Yup from 'yup'


export const ContactForm = () => {
  const bodyLimit = 300

  const formik = useFormik({
    initialValues: {
      email: '',
      name: '',
      body: '',
    },
    validationSchema: Yup.object({
      email: Yup.string()
        .email('Invalid email address')
        .required('Required'),
      name: Yup.string()
        .required('Required'),
      body: Yup.string()
        .max(bodyLimit, `Must be ${bodyLimit} characters or less`)
        .required('Required'),
    }),
    onSubmit: values => {
      alert(JSON.stringify(values, null, 2));
    },
  });
  return (
    <form onSubmit={formik.handleSubmit}>
      <label htmlFor="email">Email Address</label>
      <input
        name="email"
        {...formik.getFieldProps('email')}
      />
      {formik.touched.email && formik.errors.email && <div>{formik.errors.email}</div>}
      <label htmlFor="name">Name</label>
      <input
        name='name'
        {...formik.getFieldProps('name')}
      />
      {formik.touched.name && formik.errors.name && <div>{formik.errors.name}</div>}
      <label htmlFor="body"></label>
      <input
        name='body'
        {...formik.getFieldProps('body')}
      />
      {formik.touched.body && formik.errors.body && <div>{formik.errors.body}</div>}
      <button type="submit">Submit</button>
    </form>
  );
};


