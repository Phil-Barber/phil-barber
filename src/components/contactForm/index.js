import React from 'react';
import { Formik, Form } from 'formik';
import * as Yup from 'yup';
import * as S from './index.styled';

const URL =
  'https://8hot7pcq6e.execute-api.eu-west-2.amazonaws.com/dev/email/send';

export const ContactForm = () => {
  const bodyLimit = 300;

  return (
    <Formik
      initialValues={{
        email: '',
        name: '',
        body: '',
      }}
      validationSchema={Yup.object({
        email: Yup.string()
          .email('Invalid email address')
          .required('Required'),
        name: Yup.string().required('Required'),
        company: Yup.string(),
        body: Yup.string()
          .max(bodyLimit, `Must be ${bodyLimit} characters or less`)
          .required('Required'),
      })}
      onSubmit={(values, { setSubmitting, resetForm, setStatus }) => {
        const req = new XMLHttpRequest();
        req.open('POST', URL, true);
        req.setRequestHeader('Content-Type', 'application/json');
        req.addEventListener('load', function() {
          setSubmitting(false);
          if (req.status < 400) {
            // Success
            resetForm({});
            setStatus({ success: true });
          } else {
            setStatus({ success: false });
            throw Error('Request failed: ' + req.statusText);
          }
        });
        req.send(JSON.stringify(values));
      }}
    >
      {formik => (
        <>
          <Form>
            <S.FullWidthInput
              label="Email Address*"
              name="email"
              type="email"
            />
            <S.FlexBox>
              <S.ShortInput label="Name*" name="name" type="text" />
              <S.ShortInput label="Company" name="company" type="text" />
            </S.FlexBox>
            <S.MessageInput
              label="Message*"
              name="body"
              textarea
              placeholder={`Hi Phil,

Hope you're having a great day!

Listen to this really exciting project you could contribute to:`}
            />
            <S.FlexBox>
              <S.SubmitButton type="submit" disabled={formik.isSubmitting}>
                Submit
              </S.SubmitButton>
              <S.RequiredText>*Required field</S.RequiredText>
            </S.FlexBox>
          </Form>
        </>
      )}
    </Formik>
  );
};
