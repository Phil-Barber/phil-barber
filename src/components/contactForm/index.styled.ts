import styled from 'styled-components';
import { TextInput, Label, Input } from '../formik/textInput';

export const TextInputStyled = styled(TextInput)`
  ${Label} {
    display: block;
  }
`;

export const FullWidthInput = styled(TextInputStyled)`
  ${Input} {
    width: 100%;
  }
`;

export const ShortInput = styled(TextInputStyled)<{
  readonly isLeft?: boolean;
}>`
  width: 50%;

  ${({ isLeft, theme: { spacing } }) =>
    isLeft
      ? `padding-right: ${spacing.small}`
      : `padding-left: ${spacing.small}`};
`;

export const MessageInput = styled(FullWidthInput)`
  ${Input} {
    height: 200px;
  }
`;

export const FlexBox = styled.div`
  display: flex;
`;

export const SubmitButton = styled.button`
  flex-grow: 1;
`;

export const RequiredText = styled.div`
  font-size: 12px;
  flex-grow: 1;
  text-align: right;
`;
