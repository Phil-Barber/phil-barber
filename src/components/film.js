import React from 'react';
import Layout from './layout';

const Film = ({ children, filmTitle, filmPoster }) => (
  <Layout>
    <h1>{filmTitle}</h1>
    {children}
    <img src={filmPoster} alt={`${filmTitle} Poster`} />
  </Layout>
);

export default Film;
