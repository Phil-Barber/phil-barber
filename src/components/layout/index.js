import React from 'react';
import styled from 'styled-components';
import Headroom from 'react-headroom';
import { css } from '@emotion/core';
import { useStaticQuery, Link, graphql } from 'gatsby';

import { rhythm } from '../../utils/typography';

const MainHeader = styled.div`
  background-color: ${({ theme }) => theme.colors.primary};
  padding: ${({ theme }) => `${theme.spacing.xxSmall} ${theme.spacing.normal}`};

  h1 {
    color: ${({ theme }) => theme.colors.secondary};
    margin-bottom: 0;
  }
`;

const SecondaryHeader = styled.div`
  background-color: ${({ theme }) => theme.colors.secondary};
  color: ${({ theme }) => theme.colors.primary};
  border-bottom: 2px solid ${({ theme }) => theme.colors.tertiary};
  padding: ${({ theme }) => `
    ${theme.spacing.xxSmall} 
    ${theme.spacing.large}
    ${theme.spacing.xxxSmall}
  `};
  display: flex;
  justify-content: flex-end;
`;

const StyledLink = styled(Link)`
  text-decoration: none;
`;

const NavLink = styled(StyledLink)`
  font-size: 18px;
  color: ${({ theme }) => theme.colors.primary};
`;

const ContentContainer = styled.div`
  background-color: ${({ theme }) => theme.colors.tertiary};
  color: ${({ theme }) => theme.colors.secondary};
`;

const Content = styled.div`
  margin: 0 auto;
  max-width: 750px;
  padding-top: ${({ theme }) => theme.spacing.large};
  text-align: justify;
`;

const Layout = ({ children }) => {
  const data = useStaticQuery(
    graphql`
      query {
        site {
          siteMetadata {
            title
          }
        }
      }
    `
  );
  return (
    <>
      <Headroom>
        <MainHeader>
          <StyledLink to={'/'}>
            <h1
              css={css`
                margin-bottom: ${rhythm(2)};
                font-style: normal;
              `}
            >
              {data.site.siteMetadata.title}
            </h1>
          </StyledLink>
        </MainHeader>
        <SecondaryHeader>
          <NavLink to={'/contact/'}>
            <div>Contact</div>
          </NavLink>
        </SecondaryHeader>
      </Headroom>
      <ContentContainer>
        <Content>{children}</Content>
      </ContentContainer>
    </>
  );
};

export default Layout;
