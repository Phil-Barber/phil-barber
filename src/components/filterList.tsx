import React, { useState } from 'react';
import { Badge } from '../components/badge';

type items = React.ReactNode[];
type filters = { [key: string]: (lookup: any) => boolean };
type lookupFunc = (item: any) => string;
interface Props {
  items: items;
  filters: filters;
  lookupFunc: lookupFunc;
  children: (filtered: React.ReactNode[]) => React.ReactNode;
}

const useFilters = (
  items: items,
  filters: filters,
  lookupFunc: lookupFunc
): [Array<string>, React.ReactNode[], (clicked: string) => () => void] => {
  const [selected, setSelected] = useState<Array<string>>([]);
  const toggleSelected = (clicked: string) => () => {
    setSelected((selected) =>
      selected.includes(clicked)
        ? selected.filter((item) => item !== clicked)
        : [...selected, clicked]
    );
  };

  const filteredItems = items.filter((item) => {
    if (!selected.length) return true;

    return selected.reduce(
      (acc, key) => acc || filters[key](lookupFunc(item)),
      false
    );
  });

  return [selected, filteredItems, toggleSelected];
};

export const FilterList: React.FC<Props> = ({
  items,
  filters,
  lookupFunc,
  children,
}: Props) => {
  const [selected, filteredItems, toggleSelected] = useFilters(
    items,
    filters,
    lookupFunc
  );

  const filterButtons = Object.keys(filters).map((key) => (
    <Badge
      key={key}
      onClick={toggleSelected(key)}
      isSelected={selected.includes(key)}
    >
      {key}
    </Badge>
  ));

  return (
    <>
      {filterButtons}
      {children(filteredItems)}
    </>
  );
};
