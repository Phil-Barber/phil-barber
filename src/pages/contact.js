import React from 'react';
import PageWrapper from '../components/pageWrapper';
import { PageLayout } from '../components/styled';
import { ContactForm } from '../components/contactForm';
import SEO from '../components/seo';

export const Contact = () => (
  <PageWrapper>
    <PageLayout>
      <SEO title="contact" description="Get in contact with me" />
      <h1>Get In Touch!</h1>
      <p>
        If you&apos;ve got a social enterprise or a good cause that you would
        like some technical help with, don&apos;t hesitate to get in touch using
        the form below.
      </p>
      <ContactForm />
    </PageLayout>
  </PageWrapper>
);

export default Contact;
