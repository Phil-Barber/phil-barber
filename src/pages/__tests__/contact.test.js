import React from 'react';
import { render } from '../../../testUtils';
import { Contact } from '../contact';
import * as form from '../../components/contactForm';

describe('Contact', () => {
  jest
    .spyOn(form, 'ContactForm')
    .mockImplementation(() => <div>Mock Form</div>);

  it('Matches snapshot', () => {
    const { container } = render(<Contact />);
    expect(container).toMatchSnapshot();
  });
});
