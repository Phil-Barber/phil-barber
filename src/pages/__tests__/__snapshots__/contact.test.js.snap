// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Matches snapshot 1`] = `
<div>
  <div
    class="headroom-wrapper"
  >
    <div
      class="headroom headroom--unfixed"
      style="position: relative; top: 0px; left: 0px; right: 0px; z-index: 1; transform: translate3D(0, 0, 0);"
    >
      <div
        class="sc-AxjAm fEzfgO"
      >
        <a
          class="sc-AxgMl ZJAph"
          href="/"
        >
          <h1>
            Test title
          </h1>
        </a>
        <button
          class="sc-AxhUy blLbLW"
        >
          Let it Snow! ðŸ‘ˆ
        </button>
        <span
          class="sc-AxirZ gtVPNy"
          style="transform: rotate(0deg);"
        >
          &gt;
        </span>
      </div>
    </div>
  </div>
  <div
    class="sc-AxiKw jmqPVm sc-AxmLO gkfGJy"
  >
    <div
      class="sc-Axmtr bdeNGe"
    >
      <div
        class="sc-fzozJi lhSHn"
      >
        <h1>
          Get In Touch!
        </h1>
        <p>
          If you've got a social enterprise or a good cause that you would like some technical help with, don't hesitate to get in touch using the form below.
        </p>
        <div>
          Mock Form
        </div>
      </div>
    </div>
    <div
      class="sc-AxhCb bdrWTY"
    />
  </div>
</div>
`;
